// @voiceflow/base-types/build/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Version = exports.Trace = exports.Text = exports.Request = exports.Project = exports.Node = exports.Models = exports.Button = void 0;
exports.Button = __importStar(require("./button"));
exports.Models = __importStar(require("./models"));
exports.Node = __importStar(require("./node"));
exports.Project = __importStar(require("./project"));
exports.Request = __importStar(require("./request"));
exports.Text = __importStar(require("./text"));
exports.Trace = __importStar(require("./trace"));
exports.Version = __importStar(require("./version"));
