// @voiceflow/runtime/build/lib/Program/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProgramModel = void 0;
class ProgramModel {
    constructor({ id, lines, variables = [], commands = [], startId }) {
        this.commands = [];
        this.variables = [];
        this.id = id;
        this.nodes = lines;
        this.commands = commands;
        this.variables = variables;
        this.startNodeID = startId;
    }
    getID() {
        return this.id;
    }
    getNode(nodeID) {
        // eslint-disable-next-line no-prototype-builtins
        if (!(nodeID && this.nodes.hasOwnProperty(nodeID))) {
            return null;
        }
        return Object.assign(Object.assign({}, this.nodes[nodeID]), { id: nodeID });
    }
    getCommands() {
        return this.commands;
    }
    getStartNodeID() {
        return this.startNodeID;
    }
    getVariables() {
        return this.variables;
    }
    getRaw() {
        return this.nodes;
    }
}
exports.ProgramModel = ProgramModel;
exports.default = ProgramModel;
