// @voiceflow/api-sdk/build/resources/base.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const s = __importStar(require("superstruct"));
const utils_1 = require("../utils");
class BaseResource {
    constructor({ fetch, clazz, schema, modelIDKey, resourceEndpoint, postPutExcludedFields = [], }) {
        this.fetch = fetch;
        this.clazz = clazz;
        this.modelIDKey = modelIDKey;
        this.resourceEndpoint = resourceEndpoint;
        this.struct = s.object(schema);
        this.patchStruct = s.partial(schema);
        this.putAndPostStruct = utils_1.createPutAndPostStruct(schema, modelIDKey, postPutExcludedFields);
    }
    _getEndpoint() {
        return this.resourceEndpoint;
    }
    _getFieldsQuery(fields) {
        return fields ? `?fields=${fields.join(',')}` : '';
    }
    _assertModelID(id) {
        s.assert(id, this.struct.schema[this.modelIDKey]);
    }
    _assertPatchBody(body) {
        s.assert(body, this.patchStruct);
    }
    _assertPutAndPostBody(body) {
        s.assert(body, this.putAndPostStruct);
    }
    options(options) {
        const { clazz: Clazz } = this;
        return new Clazz(this.fetch.initWithOptions(options));
    }
}
exports.default = BaseResource;
