// @voiceflow/api-sdk/build/resources/diagram.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.modelIDKey = void 0;
const s = __importStar(require("superstruct"));
const models_1 = require("../models");
const utils_1 = require("../utils");
const crud_1 = __importDefault(require("./crud"));
const ENDPOINT = 'diagrams';
exports.modelIDKey = '_id';
class DiagramResource extends crud_1.default {
    constructor(fetch) {
        super({
            fetch,
            clazz: DiagramResource,
            schema: models_1.SDiagram.schema,
            modelIDKey: exports.modelIDKey,
            resourceEndpoint: ENDPOINT,
        });
        this._nodePutAndPostStruct = utils_1.createPutAndPostStruct(models_1.SNode.schema, 'id', [], true);
    }
    async get(id, fields) {
        return fields ? super._getByID(id, fields) : super._getByID(id);
    }
    async getRTC(id) {
        const { data } = await this.fetch.get(`${this._getCRUDEndpoint(id)}/rtc`);
        return data;
    }
    async create(body) {
        return super._post(body);
    }
    async update(id, body) {
        return super._patch(id, body);
    }
    async updateNode(id, nodeID, body) {
        this._assertModelID(id);
        s.assert(nodeID, models_1.SNodeID);
        s.assert(body, this._nodePutAndPostStruct);
        const { data } = await this.fetch.put(`${this._getCRUDEndpoint(id)}/nodes/${nodeID}`, body);
        return data;
    }
    async delete(id) {
        return super._delete(id);
    }
}
exports.default = DiagramResource;
