// @voiceflow/api-sdk/build/resources/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var user_1 = require("./user");
Object.defineProperty(exports, "User", { enumerable: true, get: function () { return user_1.default; } });
var apiKey_1 = require("./apiKey");
Object.defineProperty(exports, "APIKey", { enumerable: true, get: function () { return apiKey_1.default; } });
var project_1 = require("./project");
Object.defineProperty(exports, "Project", { enumerable: true, get: function () { return project_1.default; } });
var version_1 = require("./version");
Object.defineProperty(exports, "Version", { enumerable: true, get: function () { return version_1.default; } });
var program_1 = require("./program");
Object.defineProperty(exports, "Program", { enumerable: true, get: function () { return program_1.default; } });
var diagram_1 = require("./diagram");
Object.defineProperty(exports, "Diagram", { enumerable: true, get: function () { return diagram_1.default; } });
var prototypeProgram_1 = require("./prototypeProgram");
Object.defineProperty(exports, "PrototypeProgram", { enumerable: true, get: function () { return prototypeProgram_1.default; } });
