// @voiceflow/api-sdk/build/resources/program.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.modelIDKey = void 0;
const models_1 = require("../models");
const crud_1 = __importDefault(require("./crud"));
const ENDPOINT = 'programs';
exports.modelIDKey = 'id';
class ProgramResource extends crud_1.default {
    constructor(fetch, { resourceEndpoint = ENDPOINT } = {}) {
        super({
            fetch,
            clazz: ProgramResource,
            schema: models_1.SProgram.schema,
            modelIDKey: exports.modelIDKey,
            resourceEndpoint,
        });
    }
    async get(id, fields) {
        return fields ? super._getByID(id, fields) : super._getByID(id);
    }
    async create(body) {
        return super._post(body);
    }
    async update(id, body) {
        return super._put(id, body);
    }
    async delete(id) {
        return super._delete(id);
    }
}
exports.default = ProgramResource;
