// @voiceflow/api-sdk/build/resources/user.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseJWT = void 0;
const atob_1 = __importDefault(require("atob"));
const utils_1 = require("../utils");
exports.parseJWT = (token) => {
    var _a;
    const base64Url = token.split('.')[1];
    const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
    const jsonPayload = decodeURIComponent((((_a = utils_1.getWindow()) === null || _a === void 0 ? void 0 : _a.atob) || atob_1.default)(base64)
        .split('')
        .map((c) => `%${`00${c.charCodeAt(0).toString(16)}`.slice(-2)}`)
        .join(''));
    return JSON.parse(jsonPayload);
};
class User {
    constructor(authorization) {
        this.creatorID = 0;
        this.name = '';
        this.email = '';
        if (!this.isAPIKey(authorization)) {
            const { id, name, email } = exports.parseJWT(authorization);
            this.creatorID = id;
            this.name = name;
            this.email = email;
        }
    }
    isAPIKey(authorization) {
        return authorization.startsWith('VF.');
    }
}
exports.default = User;
