// @voiceflow/api-sdk/build/models/project.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SProject = exports.ProjectPrivacy = exports.SProjectPrototype = exports.SProjectPrototypeNLP = exports.SProjectPrototypeLuis = exports.ProjectLinkType = exports.ProjectPrototypeNLPType = void 0;
const s = __importStar(require("superstruct"));
const member_1 = require("./member");
const shared_1 = require("./shared");
var ProjectPrototypeNLPType;
(function (ProjectPrototypeNLPType) {
    ProjectPrototypeNLPType["LUIS"] = "LUIS";
})(ProjectPrototypeNLPType = exports.ProjectPrototypeNLPType || (exports.ProjectPrototypeNLPType = {}));
var ProjectLinkType;
(function (ProjectLinkType) {
    ProjectLinkType["CURVED"] = "CURVED";
    ProjectLinkType["STRAIGHT"] = "STRAIGHT";
})(ProjectLinkType = exports.ProjectLinkType || (exports.ProjectLinkType = {}));
exports.SProjectPrototypeLuis = s.object({
    type: s.enums([ProjectPrototypeNLPType.LUIS]),
    appID: s.string(),
    resourceID: s.optional(s.string()),
});
exports.SProjectPrototypeNLP = s.union([exports.SProjectPrototypeLuis]);
exports.SProjectPrototype = s.object({
    nlp: s.optional(exports.SProjectPrototypeNLP),
    data: s.object(),
    trainedModel: s.optional(shared_1.SPrototypeModel),
    lastTrainedTime: s.optional(s.number()),
});
var ProjectPrivacy;
(function (ProjectPrivacy) {
    ProjectPrivacy["PUBLIC"] = "public";
    ProjectPrivacy["PRIVATE"] = "private";
})(ProjectPrivacy = exports.ProjectPrivacy || (exports.ProjectPrivacy = {}));
exports.SProject = s.object({
    _id: shared_1.SProjectID,
    teamID: shared_1.STeamID,
    creatorID: shared_1.SCreatorID,
    name: shared_1.SName,
    image: s.optional(s.string()),
    members: s.array(member_1.SMember),
    privacy: s.optional(s.enums([ProjectPrivacy.PRIVATE, ProjectPrivacy.PUBLIC])),
    platform: shared_1.SPlatform,
    linkType: s.optional(s.enums(Object.values(ProjectLinkType))),
    prototype: s.optional(exports.SProjectPrototype),
    devVersion: s.optional(shared_1.SVersionID),
    liveVersion: s.optional(shared_1.SVersionID),
    platformData: shared_1.SBasePlatformData,
});
