// @voiceflow/runtime/build/lib/DataAPI/creatorDataAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const api_sdk_1 = __importDefault(require("@voiceflow/api-sdk"));
class CreatorDataAPI {
    constructor({ endpoint, authorization, clientKey = '', prototype = true, }, VFClient = api_sdk_1.default) {
        this.init = async () => {
            // no-op
        };
        this.fetchDisplayById = async () => {
            return null;
        };
        this.getProgram = async (programID) => {
            if (this.prototype) {
                return (await this.client.prototypeProgram.get(programID));
            }
            return (await this.client.program.get(programID));
        };
        this.getVersion = async (versionID) => {
            return (await this.client.version.get(versionID));
        };
        this.getProject = async (projectID) => {
            return (await this.client.project.get(projectID));
        };
        this.client = new VFClient({ apiEndpoint: endpoint, clientKey }).generateClient({ authorization });
        this.prototype = prototype;
    }
}
exports.default = CreatorDataAPI;
