// @voiceflow/runtime/build/lib/DataAPI/serverDataAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class ServerDataAPI {
    constructor({ platform, adminToken, dataEndpoint }, { axios }) {
        this.init = async () => {
            const { data: { token: VF_DATA_SECRET }, } = await this.axios.post(`${this.dataEndpoint}/generate-platform-token`, { platform: this.platform, ttl_min: 525600 }, { headers: { admintoken: this.adminToken } });
            this.client = this.axios.create({
                baseURL: this.dataEndpoint,
                headers: { authorization: `Bearer ${VF_DATA_SECRET}` },
            });
        };
        this.fetchDisplayById = async (displayId) => {
            const { data } = await this.client.get(`/metadata/displays/${displayId}`);
            return data !== null && data !== void 0 ? data : null;
        };
        this.getProgram = async (programID) => {
            const { data } = await this.client.get(`/diagrams/${programID}`);
            return data;
        };
        this.getVersion = async (versionID) => {
            const { data } = await this.client.get(`/version/${versionID}`);
            return data;
        };
        this.getProject = async (projectID) => {
            const { data } = await this.client.get(`/project/${projectID}`);
            return data;
        };
        this.axios = axios;
        this.platform = platform;
        this.adminToken = adminToken;
        this.dataEndpoint = dataEndpoint;
    }
}
exports.default = ServerDataAPI;
