// @voiceflow/general-types/build/constants/base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Language = exports.Locale = exports.Voice = void 0;
var Voice;
(function (Voice) {
    Voice["AUDIO"] = "audio";
    Voice["DEFAULT"] = "default";
})(Voice = exports.Voice || (exports.Voice = {}));
var Locale;
(function (Locale) {
    // English
    Locale["EN_US"] = "en-US";
    // Arabic
    Locale["AR_AR"] = "ar-AR";
    // Chinese
    Locale["ZH_CN"] = "zh-CN";
    // Dutch
    Locale["NL_NL"] = "nl-NL";
    // French
    Locale["FR_FR"] = "fr-FR";
    Locale["FR_CA"] = "fr-CA";
    // German
    Locale["DE_DE"] = "de-DE";
    // Gujarati
    Locale["GU_IN"] = "gu-IN";
    // Hindi
    Locale["HI_IN"] = "hi-IN";
    // Italian
    Locale["IT_IT"] = "it-IT";
    // Japanese
    Locale["JA_JP"] = "ja-JP";
    // Korean
    Locale["KO_KR"] = "ko-KR";
    // Marathi
    Locale["MR_IN"] = "mr-IN";
    // Portugese
    Locale["PT_BR"] = "pt-BR";
    // Spanish
    Locale["ES_ES"] = "es-ES";
    Locale["ES_MX"] = "es-MX";
    // Tamil
    Locale["TA_IN"] = "ta-IN";
    // Telugu
    Locale["TE_IN"] = "te-IN";
    // Turkish
    Locale["TR_TR"] = "tr-TR";
})(Locale = exports.Locale || (exports.Locale = {}));
var Language;
(function (Language) {
    // English
    Language["EN"] = "en";
    // Arabic
    Language["AR"] = "ar";
    // Chinese
    Language["ZH"] = "zh";
    // Dutch
    Language["NL"] = "nl";
    // French
    Language["FR"] = "fr";
    // German
    Language["DE"] = "de";
    // Gujarati
    Language["GU"] = "gu";
    // Hindi
    Language["HI"] = "hi";
    // Italian
    Language["IT"] = "it";
    // Japanese
    Language["JA"] = "ja";
    // Korean
    Language["KO"] = "ko";
    // Marathi
    Language["MR"] = "mr";
    // Portugese
    Language["PT"] = "pt";
    // Spanish
    Language["ES"] = "es";
    // Tamil
    Language["TA"] = "ta";
    // Telugu
    Language["TE"] = "te";
    // Turkish
    Language["TR"] = "tr";
})(Language = exports.Language || (exports.Language = {}));
