// @voiceflow/base-types/build/request/index.js
"use strict";
/* eslint-disable @typescript-eslint/no-empty-interface */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.isGeneralRequest = exports.isIntentRequest = exports.isLaunchRequest = exports.isActionRequest = exports.isTextRequest = exports.RequestType = exports.Action = void 0;
exports.Action = __importStar(require("./action"));
var RequestType;
(function (RequestType) {
    RequestType["TEXT"] = "text";
    RequestType["ACTION"] = "action";
    RequestType["INTENT"] = "intent";
    RequestType["LAUNCH"] = "launch";
})(RequestType = exports.RequestType || (exports.RequestType = {}));
const isTextRequest = (request) => request.type === RequestType.TEXT;
exports.isTextRequest = isTextRequest;
const isActionRequest = (request) => request.type === RequestType.ACTION;
exports.isActionRequest = isActionRequest;
const isLaunchRequest = (request) => request.type === RequestType.LAUNCH;
exports.isLaunchRequest = isLaunchRequest;
const isIntentRequest = (request) => request.type === RequestType.INTENT;
exports.isIntentRequest = isIntentRequest;
const ALL_REQUEST_TYPES = Object.values(RequestType);
const isGeneralRequest = (request) => !ALL_REQUEST_TYPES.includes(request.type);
exports.isGeneralRequest = isGeneralRequest;
