// @voiceflow/api-sdk/build/resources/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Version = exports.User = exports.PrototypeProgram = exports.Project = exports.Program = exports.Diagram = exports.APIKey = exports.Analytics = void 0;
var analytics_1 = require("./analytics");
Object.defineProperty(exports, "Analytics", { enumerable: true, get: function () { return __importDefault(analytics_1).default; } });
var apiKey_1 = require("./apiKey");
Object.defineProperty(exports, "APIKey", { enumerable: true, get: function () { return __importDefault(apiKey_1).default; } });
var diagram_1 = require("./diagram");
Object.defineProperty(exports, "Diagram", { enumerable: true, get: function () { return __importDefault(diagram_1).default; } });
var program_1 = require("./program");
Object.defineProperty(exports, "Program", { enumerable: true, get: function () { return __importDefault(program_1).default; } });
var project_1 = require("./project");
Object.defineProperty(exports, "Project", { enumerable: true, get: function () { return __importDefault(project_1).default; } });
var prototypeProgram_1 = require("./prototypeProgram");
Object.defineProperty(exports, "PrototypeProgram", { enumerable: true, get: function () { return __importDefault(prototypeProgram_1).default; } });
var user_1 = require("./user");
Object.defineProperty(exports, "User", { enumerable: true, get: function () { return __importDefault(user_1).default; } });
var version_1 = require("./version");
Object.defineProperty(exports, "Version", { enumerable: true, get: function () { return __importDefault(version_1).default; } });
