// @voiceflow/base-types/build/node/api.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.APIActionType = exports.APIMethod = exports.APIBodyType = void 0;
var APIBodyType;
(function (APIBodyType) {
    APIBodyType["FORM_DATA"] = "formData";
    APIBodyType["RAW_INPUT"] = "rawInput";
    APIBodyType["URL_ENCODED"] = "urlEncoded";
})(APIBodyType = exports.APIBodyType || (exports.APIBodyType = {}));
var APIMethod;
(function (APIMethod) {
    APIMethod["GET"] = "GET";
    APIMethod["PUT"] = "PUT";
    APIMethod["POST"] = "POST";
    APIMethod["PATCH"] = "PATCH";
    APIMethod["DELETE"] = "DELETE";
})(APIMethod = exports.APIMethod || (exports.APIMethod = {}));
var APIActionType;
(function (APIActionType) {
    APIActionType["GET"] = "Make a GET Request";
    APIActionType["PUT"] = "Make a PUT Request";
    APIActionType["POST"] = "Make a POST Request";
    APIActionType["PATCH"] = "Make a PATCH Request";
    APIActionType["DELETE"] = "Make a DELETE Request";
})(APIActionType = exports.APIActionType || (exports.APIActionType = {}));
