// @voiceflow/base-types/build/node/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeType = void 0;
var NodeType;
(function (NodeType) {
    NodeType["TEXT"] = "text";
    NodeType["SPEAK"] = "speak";
    NodeType["START"] = "start";
    NodeType["CARD"] = "card";
    NodeType["BUTTONS"] = "buttons";
    NodeType["INTERACTION"] = "interaction";
    // logic
    NodeType["SET"] = "set";
    NodeType["SET_V2"] = "setV2";
    NodeType["IF"] = "if";
    NodeType["IF_V2"] = "ifV2";
    NodeType["RANDOM"] = "random";
    NodeType["CAPTURE"] = "capture";
    // integrations
    NodeType["API"] = "api";
    NodeType["ZAPIER"] = "zapier";
    NodeType["INTEGRATIONS"] = "integrations";
    NodeType["GOOGLE_SHEETS"] = "google_sheets";
    // advanced
    NodeType["INTENT"] = "intent";
    NodeType["FLOW"] = "flow";
    NodeType["COMPONENT"] = "component";
    NodeType["CODE"] = "code";
    NodeType["EXIT"] = "exit";
    NodeType["PROMPT"] = "prompt";
    NodeType["COMMAND"] = "command";
    NodeType["TRACE"] = "trace";
    // other
    NodeType["STREAM"] = "stream";
    NodeType["VISUAL"] = "visual";
    NodeType["GENERAL"] = "general";
    NodeType["DIRECTIVE"] = "directive";
    NodeType["DEPRECATED"] = "deprecated";
})(NodeType = exports.NodeType || (exports.NodeType = {}));
