// @voiceflow/base-types/build/node/visual.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrameType = exports.VisualType = exports.APLType = exports.CanvasVisibility = exports.DeviceType = void 0;
var DeviceType;
(function (DeviceType) {
    DeviceType["MOBILE"] = "mobile";
    DeviceType["TABLET"] = "tablet";
    DeviceType["DESKTOP"] = "desktop";
    DeviceType["SMART_WATCH"] = "smart_watch";
    DeviceType["TELEVISION"] = "television";
    DeviceType["IN_CAR_DISPLAY"] = "in_car_display";
    DeviceType["ECHO_SPOT"] = "echo_spot";
    DeviceType["ECHO_SHOW_8"] = "echo_show_8";
    DeviceType["ECHO_SHOW_10"] = "echo_show_10";
    DeviceType["FIRE_HD_8"] = "fire_hd_8";
    DeviceType["FIRE_HD_10"] = "fire_hd_10";
    DeviceType["FIRE_TV_CUBE"] = "fire_tv_cube";
    DeviceType["GOOGLE_NEST_HUB"] = "google_nest_hub";
})(DeviceType = exports.DeviceType || (exports.DeviceType = {}));
var CanvasVisibility;
(function (CanvasVisibility) {
    CanvasVisibility["FULL"] = "full";
    CanvasVisibility["HIDDEN"] = "hidden";
    CanvasVisibility["CROPPED"] = "cropped";
})(CanvasVisibility = exports.CanvasVisibility || (exports.CanvasVisibility = {}));
var APLType;
(function (APLType) {
    APLType["JSON"] = "JSON";
    APLType["SPLASH"] = "SPLASH";
})(APLType = exports.APLType || (exports.APLType = {}));
var VisualType;
(function (VisualType) {
    VisualType["APL"] = "apl";
    VisualType["IMAGE"] = "image";
})(VisualType = exports.VisualType || (exports.VisualType = {}));
var FrameType;
(function (FrameType) {
    FrameType["AUTO"] = "AUTO";
    FrameType["DEVICE"] = "DEVICE";
    FrameType["CUSTOM_SIZE"] = "CUSTOM_SIZE";
})(FrameType = exports.FrameType || (exports.FrameType = {}));
