// @voiceflow/runtime/build/lib/Lifecycle/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AbstractLifecycle = exports.EventType = void 0;
const types_1 = require("./types");
Object.defineProperty(exports, "EventType", { enumerable: true, get: function () { return types_1.EventType; } });
class Lifecycle {
    constructor() {
        this.events = {};
    }
    setEvent(type, callback) {
        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
        // @ts-ignore
        this.events[type] = callback;
    }
    getEvent(type) {
        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
        // @ts-ignore
        return this.events[type];
    }
    async callEvent(type, event, runtime) {
        var _a;
        await ((_a = this.getEvent(type)) === null || _a === void 0 ? void 0 : _a(Object.assign(Object.assign({}, event), { runtime })));
    }
}
class AbstractLifecycle {
    constructor(events = new Lifecycle()) {
        this.events = events;
    }
    setEvent(type, callback) {
        this.events.setEvent(type, callback);
    }
    async callEvent(type, event, runtime) {
        await this.events.callEvent(type, event, runtime);
    }
}
exports.AbstractLifecycle = AbstractLifecycle;
exports.default = Lifecycle;
