// @voiceflow/runtime/build/lib/Lifecycle/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventType = void 0;
var EventType;
(function (EventType) {
    EventType["updateWillExecute"] = "updateWillExecute";
    EventType["updateDidExecute"] = "updateDidExecute";
    EventType["updateDidCatch"] = "updateDidCatch";
    EventType["programWillFetch"] = "programWillFetch";
    EventType["programDidFetch"] = "programDidFetch";
    EventType["stateWillExecute"] = "stateWillExecute";
    EventType["stateDidExecute"] = "stateDidExecute";
    EventType["stateDidCatch"] = "stateDidCatch";
    EventType["handlerWillHandle"] = "handlerWillHandle";
    EventType["handlerDidHandle"] = "handlerDidHandle";
    EventType["handlerDidCatch"] = "handlerDidCatch";
    EventType["stackWillChange"] = "stackWillChange";
    EventType["stackDidChange"] = "stackDidChange";
    EventType["frameDidFinish"] = "frameDidFinish";
    EventType["storageWillUpdate"] = "storageWillUpdate";
    EventType["storageDidUpdate"] = "storageDidUpdate";
    EventType["turnWillUpdate"] = "turnWillUpdate";
    EventType["turnDidUpdate"] = "turnDidUpdate";
    EventType["variablesWillUpdate"] = "variablesWillUpdate";
    EventType["variablesDidUpdate"] = "variablesDidUpdate";
    EventType["traceWillAdd"] = "traceWillAdd";
})(EventType = exports.EventType || (exports.EventType = {}));
