// @voiceflow/runtime/build/lib/Context/index.js
"use strict";
/* eslint-disable no-await-in-loop */
/* eslint-disable no-restricted-syntax */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TurnBuilder = exports.ContextBuilder = void 0;
class ContextBuilder {
    constructor() {
        this.pipes = [];
    }
    addHandlers(...handlers) {
        this.pipes.push(handlers);
        return this;
    }
    async handle(_request) {
        let request = _request;
        for (const handlers of this.pipes) {
            request.end = false;
            for (const handler of handlers) {
                request = await handler.handle(request);
                if (request.end)
                    break;
            }
        }
        return request;
    }
}
exports.ContextBuilder = ContextBuilder;
class TurnBuilder extends ContextBuilder {
    constructor(init) {
        super();
        this.init = init;
    }
    async handle(_request) {
        return super.handle(await this.init.handle(_request));
    }
    async resolve(_request) {
        const { request, state, trace } = await this.handle(_request);
        return {
            request,
            state,
            trace,
        };
    }
}
exports.TurnBuilder = TurnBuilder;
