// @voiceflow/api-sdk/build/client.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = exports.PublicClient = void 0;
const fetch_1 = __importDefault(require("./fetch"));
const resources_1 = require("./resources");
class PublicClient {
    constructor({ clientKey, apiEndpoint, authorization, options }) {
        this.fetch = new fetch_1.default({ clientKey, apiEndpoint, authorization, options });
        this.apiKey = new resources_1.APIKey(this.fetch);
        this.project = new resources_1.Project(this.fetch);
        this.version = new resources_1.Version(this.fetch);
        this.program = new resources_1.Program(this.fetch);
        this.diagram = new resources_1.Diagram(this.fetch);
        this.prototypeProgram = new resources_1.PrototypeProgram(this.fetch);
    }
}
exports.PublicClient = PublicClient;
class Client extends PublicClient {
    constructor({ clientKey, apiEndpoint, authorization, options }) {
        super({ clientKey, apiEndpoint, authorization, options });
        this.user = new resources_1.User(authorization);
    }
}
exports.Client = Client;
