// @voiceflow/api-sdk/build/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SGenerateClientParams = exports.SParams = void 0;
const s = __importStar(require("superstruct"));
const client_1 = require("./client");
__exportStar(require("./models"), exports);
exports.SParams = s.object({
    clientKey: s.string(),
    apiEndpoint: s.string(),
});
exports.SGenerateClientParams = s.type({
    authorization: s.string(),
});
class ApiSDK {
    constructor({ clientKey, apiEndpoint }) {
        s.assert({ clientKey, apiEndpoint }, exports.SParams);
        this.clientKey = clientKey;
        this.apiEndpoint = apiEndpoint;
    }
    generatePublicClient({ options } = {}) {
        return new client_1.PublicClient({
            options,
            clientKey: this.clientKey,
            apiEndpoint: this.apiEndpoint,
        });
    }
    generateClient({ authorization, options }) {
        s.assert({ authorization }, exports.SGenerateClientParams);
        return new client_1.Client({
            clientKey: this.clientKey,
            options,
            apiEndpoint: this.apiEndpoint,
            authorization,
        });
    }
}
exports.default = ApiSDK;
